### ISSUE TITLE
What is your stand / opinion on Gulp, Grunt, Brunch, Gobble and other similar task / build systems?

### ISSUE THREAD

**Issue by bucaran** [isojs/isojs#5]
Although I am a strong advocate of JavaScript all the time in both my local community and whenever I have the chance, I strongly agree with [Dave King](https://www.tildedave.com/2015/01/07/i-find-gulp-extremely-frustrating.html) that these systems are hiding too much plumbing under the hood to do something a simple shell script could do for you.

I also found [this](http://blog.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool/) article invaluable when in need to explain with real examples how only `npm` can be used to accomplish this. I have more or less given up on these popular task systems ever since.

**Comment by stagas** [comment]
I agree that too much magic is bad because it's easy to create very complex builds that have more chances of failing. Personally, I use Make for the reason that its obscure syntax kind of forces you to stay simple, but that's mostly a preference. The only issue is I'd like to have the "bin" or -g option when you npm install, decoupled from npm as a little tool that installs any js file/folder and adds it in the path, creates the necessary files depending on the platform (.cmd etc for windows) because now you just have to use npm for this and I don't necessarily want to.

**Comment by jnv** [comment]
I also recommend [this article](http://substack.net/task_automation_with_npm_run) by James Halliday (aka substack) – the really painful disadvantage though is portability if you need to run builds on Windows too.

>  The only issue is I'd like to have the "bin" or -g option when you npm install, decoupled from npm as a little tool that installs any js file/folder and adds it in the path, creates the necessary files depending on the platform (.cmd etc for windows) because now you just have to use npm for this and I don't necessarily want to.

I understand that npm now aims at [heavy decoupling](http://blog.npmjs.org/post/91303926460/npm-cli-roadmap-a-periodic-update) which will allow alternate workflows, however I'd think that `npm install` is a core function of npm; I'm not sure why you wouldn't want to use npm for this?
